sandy/core/endf6.py

```diff
@@ -2475,3 +2475,162 @@ def covariance_info(self, nubar=True, xs=True, mubar=True, chi=True):
             'chi': run_chi if chi else False,
             }
         return cov_info
+
+    def _get_xs_cov(self, mf=[31, 33], mt=range(1, 10000),
+                    mat=range(1, 10000)):
+        """
+        Extract from endf6 file the cross section/nubar covariance matrices.
+
+        Parameters
+        ----------
+        mf : `int` or `list`, optional
+            MF number. The default is [31, 33].
+        mt : `int` or `list`, optional
+            MT number. The default is range(1, 10000).
+        mat : `int` or `list`, optional
+            MAT number. The default is range(1, 10000).
+
+        Returns
+        -------
+        `sandy.CategoryCov`
+            Covariance matrix.
+
+        Examples
+        --------
+        >>> endf6 = sandy.get_endf6_file('jeff_33','xs', 922380)
+        >>> cov = endf6._get_xs_cov()
+        >>> xs_cov = sandy.XsCov.from_endf6(endf6)
+        >>> assert (cov.data == xs_cov).all().all()
```

Comment:
More test needed...at least to check the input options

Reply:
As the other branches are introduced, the tests will be updated.